sklearn/tests/test_init.py

```diff
@@ -18,3 +25,35 @@ def test_import_skl():
     # "import *" is discouraged outside of the module level, hence we
     # rely on setting up the variable above
     assert_equal(_top_import_error, None)
+
+
+def test_import_sklearn_no_warnings():
+    # Test that importing scikit-learn main modules doesn't raise any warnings.
+
+    try:
+        pkgs = pkgutil.iter_modules(path=sklearn.__path__, prefix='sklearn.')
+        import_modules = '; '.join(['import ' + modname
+                                    for _, modname, _ in pkgs
+                                    if not modname.startswith('_')])
+
+        message = subprocess.check_output(['python', '-Wdefault',
```

Comment:
we should probably do this in ``tests/test_init.py`` where we check that importing sklearn raises no warning.